packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -5945,7 +5945,7 @@ abstract class RenderObjectElement extends Element {
     int oldChildrenBottom = oldChildren.length - 1;
 
     final List<Element> newChildren = oldChildren.length == newWidgets.length ?
-        oldChildren : List<Element>.filled(newWidgets.length, _NullElement.instance);
+        <Element>[...oldChildren] : List<Element>.filled(newWidgets.length, _NullElement.instance);
```

Comment:
As far as I can see, there's actually no reason to special-case `oldChildren.length == newWidgets.length` then. Could this just be simplified to:

```suggestion
    final List<Element> newChildren = List<Element>.filled(newWidgets.length, _NullElement.instance);
```